spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcQueryMethod.java

```diff
@@ -53,13 +58,38 @@ public JdbcQueryMethod(Method method, RepositoryMetadata metadata, ProjectionFac
 	 *
 	 * @return May be {@code null}.
 	 */
+
 	@Nullable
 	String getAnnotatedQuery() {
-		return getMergedAnnotationAttribute("value");
+		String annotatedValue = getNamedQueryvalue();
+		return StringUtils.hasText(annotatedValue) ? annotatedValue : getNamedQueryName();
 	}
 
 	/**
-	 * Returns the class to be used as {@link org.springframework.jdbc.core.RowMapper}
+	 * Returns the annotated query with key value if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getNamedQueryvalue() {
+		return getMergedAnnotationAttribute("value");
+	}
+	/**
+	 * Returns the annotated query with key name if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getNamedQueryName() {
```

Comment:
This doesn't look right.
1. if a name is specified in an `@Query` annotation but no such query is found we should not fall back on the default name.
2. The method name is `getNamedQueryName` but if an explicitly named query is present, that query is returned, not its name.

Reply:
Done. the method returns null and change the name.